vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SpaceManagerTest.java

```diff
@@ -116,23 +125,32 @@ public void test_storeSpace_success() throws FileStorageException {
         Mockito.verify(spaceDisplayRepo).save((SpaceDisplay)spaceDisplay);
     }
 
-
-
-    @Test(expected = SpaceDoesNotExistException.class)
-    public void test_deleteSpaceById_whenIdIsNull() throws SpaceDoesNotExistException {
-        Mockito.doThrow(IllegalArgumentException.class)
-        .when(spaceRepo).deleteById(null);
+    @Test
+    public void test_deleteSpaceById_whenIdIsNull() throws SpaceDoesNotExistException{
+        Mockito.when(managerToTest.getSpace(spaceId)).thenReturn(null).thenReturn(null);
+        Assert.assertEquals(null, managerToTest.getSpace(null));
         managerToTest.deleteSpaceById(null);
+        Assert.assertEquals(null, managerToTest.getSpace(null));
     }
 
-    @Test(expected = SpaceDoesNotExistException.class)
-    public void test_deleteSpaceById_forNonExistentId() throws SpaceDoesNotExistException {
-        String imgId = "imgId";
-        Mockito.doThrow(EmptyResultDataAccessException.class)
-        .when(spaceRepo).deleteById(imgId);
-        managerToTest.deleteSpaceById(imgId);
+    @Test
+    public void test_deleteSpaceById_forNonExistentId() {  
+        Mockito.when(managerToTest.getSpace(spaceId)).thenReturn(null).thenReturn(null);
+        Assert.assertEquals(null, managerToTest.getSpace(spaceId));
+        managerToTest.deleteSpaceById(spaceId);
+        Assert.assertEquals(null, managerToTest.getSpace(spaceId));
     }
 
+    @Test
+    public void test_deleteSpaceById_forSuccess() {  
+        Space space = new Space();
+        Optional<Space> spaceOpt = Optional.of(space);
+        Mockito.when(spaceRepo.findById(spaceId1)).thenReturn(spaceOpt).thenReturn(null);
+        Assert.assertEquals(space, managerToTest.getSpace(spaceId1));
+        managerToTest.deleteSpaceById(spaceId1);
+        Assert.assertEquals(null, managerToTest.getSpace(spaceId1));
```

Comment:
same here. test what you expect to happen if you call the method with an existing id (e.g. delete is called on repository)

Reply:
Once the space is deleted, it is expected to return null if we try to fetch that id from the repo. That is what I am trying to test.

Reply:
but this test is for `deleteSpaceById` not `getSpace`. even if `getSpace` wouldn't return null, this method should still succeed as long as the deletion is initiated.